lasy/laser.py

```diff
@@ -237,16 +238,27 @@ def propagate(self, distance, nr_boundary=None, backend="NP", show_progress=True
                 field[:, :nr_boundary, :] *= absorb_layer_shape[::-1][None, :, None]
             self.grid.set_temporal_field(field)
 
+        # Select backend
+        if use_cupy:
+            backend = "CU"
+        else:
+            backend = "NP"
+
+        k = self.omega_1d/c            
         if self.dim == "rt":
             # Construct the propagator (check if exists)
             if not hasattr(self, "prop"):
                 spatial_axes = (self.grid.axes[0],)
                 self.prop = []
+                if use_cupy:
+                    # Move quantities to CPU to create propagator
```

Comment:
@hightower8083 Would it be possible to modify the `PropagatorResampling`, so that it can take `cupy` arrays as input?
Right now, if don't move `k` and `spatial_axes` to the CPU, I get the following error:
```
lasy/laser.py:254: in propagate
    PropagatorResampling(
.../python-3.11/lib/python3.11/site-packages/axiprop/lib.py:242: in __init__
    self.init_kr(self.Rmax, self.Nr)
.../python-3.11/lib/python3.11/site-packages/axiprop/common.py:113: in init_kr
    self.kr = self.alpha / Rmax
```
Note that I am using [this branch](https://github.com/hightower8083/axiprop/tree/add-step-on-device) of `axiprop`

Reply:
well there is no need to have `k` and `spatial_axes` on GPU -- spatial axes are used to make transverse wave-numbers `k_r` that are transferred to GPU by axiprop and `k` is used per-element in the loop as scalars